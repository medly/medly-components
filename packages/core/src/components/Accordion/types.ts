export type StaticProps = {
    Header: React.FC;
    Content: React.FC;
    Context: React.Context<AccordionContextType>;
};

export type AccordionContextType = [
    boolean /** Current active state */,
    React.Dispatch<React.SetStateAction<boolean>> /** Function to be called to toggle the active state */
];
