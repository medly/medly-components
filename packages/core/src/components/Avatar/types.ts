import { AvatarSizes } from '@medly-components/theme';
import { HTMLProps, Omit } from '@medly-components/utils';

export interface AvatarProps extends Omit<HTMLProps<HTMLDivElement>, 'size'> {
    /** Size of the avatar */
    size?: AvatarSizes;
    /** Text color */
    textColor?: string;
    /** Background color */
    bgColor?: string;
    /** Hover Text color */
    hoverTextColor?: string;
    /** Hover Background color */
    hoverBgColor?: string;
    /** Hover text shadow color*/
    hoverTextShadowColor?: string;
    /** Hover image shadow color */
    hoverImgShadowColor?: string;
    /** Pressed Background color */
    pressedBgColor?: string;
    /** Pressed Text color */
    pressedTextColor?: string;
}

export interface StyledProps extends AvatarProps {
    /** To check children is image */
    isImage?: boolean;
}
