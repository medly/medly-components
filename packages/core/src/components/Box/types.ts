import { FontVariants, Theme } from '@medly-components/theme';
import { HTMLProps } from '@medly-components/utils';

export type BoxProps = Omit<HTMLProps<HTMLDivElement>, 'size'> & { theme?: Theme } & {
    /** To be used to render any component as Button */
    as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
    /** Padding along x-axis */
    px?: number;
    /** Padding along y-axis */
    py?: number;
    /** Margin along x-axis */
    mx?: number;
    /** Margin along y-axis */
    my?: number;
    /** Background color */
    bg?: string;
    /** To be used to specify color of the text inside component */
    color?: string;
    /** To be used to specify width of the component */
    width?: string;
    /** To be used to specify height of the component */
    height?: string;
    /** To be used to specify border color, borderWidth is also required for this */
    borderColor?: string;
    /** To be used to specify border width, borderColor is also required for this */
    borderWidth?: number;
    /** To be used to change border-radius */
    borderRadius?: number;
    /** To be used to specify the display type of component */
    display?: 'flex' | 'block' | 'none' | 'inline-flex' | 'inline-block' | 'table' | 'table-row' | 'list-item' | 'initial';
    /** Set it true to show loading state */
    isLoading?: boolean;
    /** Set font variant for text inside component */
    fontVariant?: FontVariants;
    /** Set shadow for the box component */
    shadowSize?: 'S' | 'M' | 'L' | 'XL';
    /** Set shadow color for the box component */
    shadowColor?: string;
};
