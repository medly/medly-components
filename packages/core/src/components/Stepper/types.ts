import { StepperSizes } from '@medly-components/theme';
import { HTMLProps, Omit } from '@medly-components/utils';
import type { FC } from 'react';

export interface CounterProps extends Omit<HTMLProps<HTMLDivElement>, 'size'> {
    size?: StepperSizes;
    active?: boolean;
}

export interface StepProps extends CounterProps {
    step?: number;
}

export interface StepperProps extends Omit<HTMLProps<HTMLDivElement>, 'size'> {
    /** Stepper size */
    size?: StepperSizes;
    /** Active step */
    activeStep?: number;
    /** Function to be called on clicking on any step */
    onStepClick?: (step: number) => void;
}

export interface StepperStaticProps {
    Step: FC<StepProps>;
}
