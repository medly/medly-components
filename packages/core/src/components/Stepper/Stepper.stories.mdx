import { Stepper } from './Stepper';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import * as stories from './Stepper.stories';

<Meta title="Core" component={Stepper} />

# Stepper

The basic `Stepper` component displays progress through a sequence of numbered steps, which you can use to navigate as well.
You can use `Stepper.Step` to add steps in this component like mentioned below.


```typescript
<Stepper activeStep={0} size="S" onStepClick={setActiveStep}>
    <Stepper.Step>Customer Information</Stepper.Step>
    <Stepper.Step>Payment information</Stepper.Step>
</Stepper>
```

<Preview withToolbar>
    <Story name="Stepper">{stories.Basic()}</Story>
</Preview>

## Props

This component offers the following props mentioned in the table below:

<Props of={Stepper} />

## StepperTheme

This component offers the following themes mentioned in the table below:

<Props of={stories.ThemeInterface} />
