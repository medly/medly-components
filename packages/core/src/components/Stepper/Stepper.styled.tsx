import styled from 'styled-components';

export const StepperStyled = styled('div')`
    display: flex;
`;
