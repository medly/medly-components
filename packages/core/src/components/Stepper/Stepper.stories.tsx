import { defaultTheme, StepperTheme } from '@medly-components/theme';
import { select } from '@storybook/addon-knobs';
import React, { FC, useState } from 'react';
import { Stepper } from './Stepper';
import { StepperProps } from './types';

const size: StepperProps['size'][] = ['S', 'M', 'L'];

export const ThemeInterface: FC<StepperTheme> = () => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.stepper
};

export const Basic = () => {
    const [activeStep, setActiveStep] = useState(0);
    return (
        <Stepper activeStep={activeStep} size={select('Size', size, 'S')} onStepClick={setActiveStep}>
            <Stepper.Step>Customer Information</Stepper.Step>
            <Stepper.Step>Payment information</Stepper.Step>
            <Stepper.Step>Booking is confirmed!</Stepper.Step>
        </Stepper>
    );
};
