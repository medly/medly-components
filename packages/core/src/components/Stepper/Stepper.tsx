import { WithStyle } from '@medly-components/utils';
import React, { FC } from 'react';
import Step from './Step/Step';
import { StepperStyled } from './Stepper.styled';
import { StepperProps, StepperStaticProps } from './types';

const Component: FC<StepperProps> = React.memo(
    React.forwardRef((props, ref) => {
        const { size, activeStep, children, onStepClick, ...restProps } = props;
        return (
            <StepperStyled ref={ref} {...restProps}>
                {React.Children.map(children, (child, index) => {
                    return React.cloneElement(child as any, {
                        size,
                        key: index + 1,
                        step: index + 1,
                        active: activeStep === index,
                        onClick: onStepClick ? () => onStepClick(index) : undefined
                    });
                })}
            </StepperStyled>
        );
    })
);

Component.defaultProps = {
    activeStep: 0
};
Component.displayName = 'Stepper';
export const Stepper: FC<StepperProps> & WithStyle & StepperStaticProps = Object.assign(Component, {
    Style: StepperStyled,
    Step
});
