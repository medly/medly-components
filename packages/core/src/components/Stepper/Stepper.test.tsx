import { fireEvent, render } from '@test-utils';
import React from 'react';
import { Stepper } from './Stepper';
import { StepperProps } from './types';

const renderer = (props: StepperProps = {}) =>
    render(
        <Stepper {...props}>
            <Stepper.Step>Customer Information</Stepper.Step>
            <Stepper.Step>Payment information</Stepper.Step>
            <Stepper.Step>Booking is confirmed!</Stepper.Step>
        </Stepper>
    );

describe('Stepper component', () => {
    it('should render correctly with all the default props', () => {
        const { container } = renderer();
        expect(container).toMatchSnapshot();
    });

    it('should render correctly with all the props given', () => {
        const mockOnStepClick = jest.fn();
        const { container } = renderer({ activeStep: 0, size: 'S', onStepClick: mockOnStepClick });
        expect(container).toMatchSnapshot();
    });

    it('should change active step when click on any step', async () => {
        const mockOnStepClick = jest.fn();
        const { container, getByText } = renderer({ activeStep: 0, size: 'S', onStepClick: mockOnStepClick });
        fireEvent.click(getByText('Payment information'));
        expect(mockOnStepClick).toBeCalledWith(1);
        expect(container).toMatchSnapshot();
    });
});
