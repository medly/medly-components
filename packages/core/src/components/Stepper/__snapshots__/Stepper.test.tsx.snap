// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper component should change active step when click on any step 1`] = `
.c5 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  text-transform: uppercase;
  display: inline-block;
}

.c6 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c3 {
  display: inline-block;
  text-align: center;
  border-radius: 50%;
  overflow: hidden;
  width: 30px;
  height: 30px;
  line-height: 30px;
  background: #012040;
}

.c3 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #ffffff;
  line-height: 30px;
  font-size: 1.4rem;
}

.c3 + .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 2.2rem;
  font-size: 1.4rem;
  color: #012040;
}

.c7 {
  display: inline-block;
  text-align: center;
  border-radius: 50%;
  overflow: hidden;
  width: 30px;
  height: 30px;
  line-height: 30px;
  background: #C7D0D8;
}

.c7 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #ffffff;
  line-height: 30px;
  font-size: 1.4rem;
}

.c7 + .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 2.2rem;
  font-size: 1.4rem;
  color: #C7D0D8;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 20px;
}

.c1 .c2 {
  margin-right: 10px;
}

.c1 * {
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      >
        <strong
          class="c4 c5"
        >
          1
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Customer Information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
      >
        <strong
          class="c4 c5"
        >
          2
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Payment information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
      >
        <strong
          class="c4 c5"
        >
          3
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Booking is confirmed!
      </strong>
    </div>
  </div>
</div>
`;

exports[`Stepper component should render correctly with all the default props 1`] = `
.c5 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  text-transform: uppercase;
  display: inline-block;
}

.c6 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c3 {
  display: inline-block;
  text-align: center;
  border-radius: 50%;
  overflow: hidden;
  width: 30px;
  height: 30px;
  line-height: 30px;
  background: #012040;
}

.c3 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #ffffff;
  line-height: 30px;
  font-size: 1.4rem;
}

.c3 + .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 2.2rem;
  font-size: 1.4rem;
  color: #012040;
}

.c7 {
  display: inline-block;
  text-align: center;
  border-radius: 50%;
  overflow: hidden;
  width: 30px;
  height: 30px;
  line-height: 30px;
  background: #C7D0D8;
}

.c7 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #ffffff;
  line-height: 30px;
  font-size: 1.4rem;
}

.c7 + .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 2.2rem;
  font-size: 1.4rem;
  color: #C7D0D8;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 20px;
}

.c1 .c2 {
  margin-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      >
        <strong
          class="c4 c5"
        >
          1
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Customer Information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
      >
        <strong
          class="c4 c5"
        >
          2
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Payment information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
      >
        <strong
          class="c4 c5"
        >
          3
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Booking is confirmed!
      </strong>
    </div>
  </div>
</div>
`;

exports[`Stepper component should render correctly with all the props given 1`] = `
.c5 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  text-transform: uppercase;
  display: inline-block;
}

.c6 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c3 {
  display: inline-block;
  text-align: center;
  border-radius: 50%;
  overflow: hidden;
  width: 30px;
  height: 30px;
  line-height: 30px;
  background: #012040;
}

.c3 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #ffffff;
  line-height: 30px;
  font-size: 1.4rem;
}

.c3 + .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 2.2rem;
  font-size: 1.4rem;
  color: #012040;
}

.c7 {
  display: inline-block;
  text-align: center;
  border-radius: 50%;
  overflow: hidden;
  width: 30px;
  height: 30px;
  line-height: 30px;
  background: #C7D0D8;
}

.c7 .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #ffffff;
  line-height: 30px;
  font-size: 1.4rem;
}

.c7 + .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 2.2rem;
  font-size: 1.4rem;
  color: #C7D0D8;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 20px;
}

.c1 .c2 {
  margin-right: 10px;
}

.c1 * {
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2 c3"
      >
        <strong
          class="c4 c5"
        >
          1
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Customer Information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
      >
        <strong
          class="c4 c5"
        >
          2
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Payment information
      </strong>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2 c7"
      >
        <strong
          class="c4 c5"
        >
          3
        </strong>
      </div>
      <strong
        class="c4 c6"
      >
        Booking is confirmed!
      </strong>
    </div>
  </div>
</div>
`;
